internal/server/http/server.go

```diff
@@ -155,14 +156,25 @@ func (s *Server) addImposterHandler(imposters []Imposter, imposterFilePath strin
 	}
 }
 
-func (s Server) unmarshalImposters(imposterFileName string, imposters *[]Imposter) error {
-	imposterFile, _ := os.Open(imposterFileName)
+func (s *Server) unmarshalImposters(imposterConfig ImposterConfig, imposters *[]Imposter) error {
+	imposterFile, _ := os.Open(imposterConfig.FilePath)
 	defer imposterFile.Close()
 
 	bytes, _ := ioutil.ReadAll(imposterFile)
-	if err := json.Unmarshal(bytes, imposters); err != nil {
-		return fmt.Errorf("%w: error while unmarshall imposter file %s", err, imposterFileName)
+
+	var parseError error
+
+	switch imposterConfig.Type {
```

Comment:
I'd be defensive here and I'd add a `default` clause here with a proper error. Otherwise, if something weird happens, we'll omit the error which is something not desired in terms of UX.